persistence/sql/migrations/sql/20230614000001000000_hydra_login_challenge_format.up.sql

```diff
@@ -0,0 +1,5 @@
+ALTER TABLE "selfservice_login_flows" DROP COLUMN "oauth2_login_challenge";
+ALTER TABLE "selfservice_registration_flows" DROP COLUMN "oauth2_login_challenge";
+
+ALTER TABLE "selfservice_login_flows" ADD COLUMN "oauth2_login_challenge_data" TEXT NULL;
```

Comment:
I'll add a test.

> Also note that this will cause users who are logging in during the migration will see an error and need to redo the flow (e.g. fill out registration form again).

Do you think this is acceptable, or should I write a Go migration that copies the values over from the old column? Or can we do this directly in SQL with an `UPDATE` statement?

Reply:
I would expect cockroach to be able to convert varchar to text itself, not sure if you need to drop the column. An update statement could also work. It will probably lock the table though for maybe a minute or so as all rows need to be updated.

Reply:
In Cockroach the column is a `UUID`, and UUIDs can only convert to `BYTES` automatically. See https://www.cockroachlabs.com/docs/v23.1/uuid#supported-casting-and-conversion.